### PR TITLE
Implement role for 14-sles

### PR DESCRIPTION
**Pull request by pierreoxs**

### CODE REVIEW

File: roles/maintenance_14_sles/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    msg: "System is Registered"
+  when: "'Not Registered' not in suseconnect_status_output.stdout"
+
+- <<: *task
+  vars:
+    taskid: 14-002
+    name: "Check if the system is properly registered at SUSE SUSEConnect --status-text"
+  ansible.builtin.debug:
+    msg: "System is Not Registered"
+  when: "'Not Registered' in suseconnect_status_output.stdout"
```
Comment (s3lph) [inline roles/maintenance_14_sles/tasks/main.yml:54]: I'd remove the two debug tasks and simply change the command task directly emit the changed status:
```suggestion
  changed_when: "'Not Registered' in suseconnect_status_output.stdout"
```
That way we don't have an unnecessary indirection (first save the output, then print something else based on it), and we get the actual output of SUSEConnect back.

File: roles/maintenance_14_sles/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  changed_when: false
+  failed_when: false
+
+- <<: *task
+  vars:
+    taskid: 14-003
+    name: "Zypper: Update package lists and check for errors zypper refresh"
+  ansible.builtin.debug:
+    msg: "Package lists updated successfully."
+  when: zypper_refresh_output.rc == 0
```
Comment (s3lph) [inline roles/maintenance_14_sles/tasks/main.yml:71]: What purpose does this task serve? This never gets printed, as it is neither changed nor failed.
I'd much rather expect the playbook to fail if the rc is nonzero.

**Review by s3lph — CHANGES_REQUESTED** [review]
Please review the way this project (ab-)uses Ansible task results:
https://github.com/adfinis/ansible-collection-maintenance/blob/main/README.md?plain=1#L10-L16

(Yes, I know it can be a bit counterintuitive...)

**Comment by pierreoxs** [comment]
@s3lph A new commit is available for review

**Comment by s3lph** [comment]
@pierreoxs you can simply use the "Re-request review" button at the top right:
image

**Review by s3lph — CHANGES_REQUESTED** [review]

**Comment by pierreoxs** [comment]
Hey @s3lph , new commit available and I used the "Re-request review" button :)

File: roles/maintenance_14_sles/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  changed_when: false
+  failed_when: false
+
+- <<: *task
+  vars:
+    taskid: 14-001
+    name: "Apparmor: Check if apparmor is enabled and enforcing apparmor_status"
+  ansible.builtin.debug:
+    msg: "{{ apparmor_status_output.stdout }}"
+  when: apparmor_status_output is succeeded
```
Comment (s3lph) [inline roles/maintenance_14_sles/tasks/main.yml:30]: On doing some [additional checking](https://manpages.ubuntu.com/manpages/focal/en/man8/aa-status.8.html), you can simply check the rc of aa-status.
```suggestion
  changed_when: "task.rc != 0"
  failed_when: "task.rc == 4"  # Insufficient permission to check AA policies.
```

File: roles/maintenance_14_sles/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+- <<: *task
+  vars:
+    taskid: 14-003
+    name: "Zypper: Update package lists and check for errors zypper refresh"
+  community.general.zypper:
+    name: '*'
+    update_cache: true
+  register: zypper_refresh_output
+  failed_when: "{{ zypper_refresh_output.rc != 0 }}"
```
Comment (s3lph) [inline roles/maintenance_14_sles/tasks/main.yml:48]: If you're not using the result of the task in another task, you don't actually need to register the task result again. All tasks using the `*task` "template" already register their result in the `task` variable, which can be used in e.g. `changed_when` or `failed_when`.

```suggestion
  failed_when: "task.rc != 0"
```

**Review by s3lph — CHANGES_REQUESTED** [review]
Only these two minor things, then I think this can be merged.

File: roles/maintenance_14_sles/tasks/main.yml (hunk 4 above)
Comment (s3lph) [reply roles/maintenance_14_sles/tasks/main.yml:48]: Wait, there actually is no `.rc` at all in the result of `community.general.zypper`. Have you tested this code?
I think `failed_when` can be omitted entirely in this case, just letting it fail when zypper says it failed.
```suggestion
```

**Review by s3lph — APPROVED** [review]

**Comment by s3lph** [comment]
Released as 0.1.24.

**Pull request merged**